File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums: return 0
        n = len(nums)
        nums.sort()

        ret = 1
        visited = {}
        for i in range(n):
            target = nums[i] - 1
            if nums[i] not in visited:
                visited[nums[i]] = 1
            if target in visited:
                visited[nums[i]] = max(visited[nums[i]], visited[target] + 1)
            ret = max(ret, visited[nums[i]])
        return ret

File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> visited;
        for( int i=0; i<nums.size(); i++ ) {
            int diff = target - nums[i];
            if( visited.find(diff) != visited.end() ) {
                return {visited[diff], i};
            }
            visited[nums[i]] = i;
        }
        return {};
    }
};
